src/offline/mod_offline_rdbms.erl

```diff
@@ -58,123 +127,84 @@ pop_messages(#jid{} = To) ->
             {error, Reason}
     end.
 
+%% Fetch messages for GDPR
+%% User and Server are not normalized
 fetch_messages(#jid{} = To) ->
     US = {LUser, LServer} = jid:to_lus(To),
-    TimeStamp = erlang:timestamp(),
-    SUser = mongoose_rdbms:escape_string(LUser),
-    SServer = mongoose_rdbms:escape_string(LServer),
-    STimeStamp = encode_timestamp(TimeStamp),
-    case rdbms_queries:fetch_offline_messages(LServer, SUser, SServer, STimeStamp) of
-        {selected, Rows} ->
-            {ok, rows_to_records(US, To, Rows)};
-        {error, Reason} ->
-            {error, Reason}
-    end.
-
-rows_to_records(US, To, Rows) ->
-    [row_to_record(US, To, Row) || Row <- Rows].
-
-row_to_record(US, To, {STimeStamp, SFrom, SPacket, SPermanentFields}) ->
-    {ok, Packet} = exml:parse(SPacket),
-    TimeStamp = usec:to_now(mongoose_rdbms:result_to_integer(STimeStamp)),
-    From = jid:from_binary(SFrom),
-    PermanentFields = extract_permanent_fields(SPermanentFields),
-    #offline_msg{us = US,
-             timestamp = TimeStamp,
-             expire = never,
-             from = From,
-             to = To,
-             packet = Packet,
-             permanent_fields = PermanentFields}.
-
-extract_permanent_fields(null) ->
-    []; %% This is needed in transition period when upgrading to MongooseIM above 3.5.0
```

Comment:
Why remove this clause? It's missing in line 206. If so, I guess we'll need to add it to the migration guide. How not to forget to do it?

Reply:
there is no test for it, so if we keep that, we would need to add one.

Reply:
put it back too, should be outdated.